From 32fe63d091ae3685c42b524cc476f4e408c2e778 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timur=20Krist=C3=B3f?= <timur.kristof@gmail.com>
Date: Sat, 15 Nov 2025 14:21:21 +0100
Subject: [PATCH] ac/gpu_info: Disable sparse VM mappings pre-Polaris, for now
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Disable sparse mappings on GFX7-8 due to GPU hangs in the VK CTS,
except Polaris where it happens to work "well enough" to pass
the VK CTS and run some games already.

Cc: mesa-stable
Signed-off-by: Timur Kristóf <timur.kristof@gmail.com>
Reviewed-by: Marek Olšák <marek.olsak@amd.com>
Reviewed-by: Samuel Pitoiset <samuel.pitoiset@gmail.com>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/38304>
(cherry picked from commit 567e1b56ef9de920401f834a3f595faf978f9bc0)
---
 .pick_status.json            | 2 +-
 src/amd/common/ac_gpu_info.c | 5 ++++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/.pick_status.json b/.pick_status.json
index 63f71cfa91c79..3ce361ae32e65 100644
--- a/.pick_status.json
+++ b/.pick_status.json
@@ -584,7 +584,7 @@
         "description": "ac/gpu_info: Disable sparse VM mappings pre-Polaris, for now",
         "nominated": true,
         "nomination_type": 1,
-        "resolution": 0,
+        "resolution": 1,
         "main_sha": null,
         "because_sha": null,
         "notes": null
diff --git a/src/amd/common/ac_gpu_info.c b/src/amd/common/ac_gpu_info.c
index 77931a636cf26..fc3ef1d739f41 100644
--- a/src/amd/common/ac_gpu_info.c
+++ b/src/amd/common/ac_gpu_info.c
@@ -739,8 +739,11 @@ ac_query_gpu_info(int fd, void *dev_p, struct radeon_info *info,
    info->has_eqaa_surface_allocator = info->gfx_level < GFX11;
    /* Disable sparse mappings on GFX6 due to VM faults in CP DMA. Enable them once
     * these faults are mitigated in software.
+    * Disable sparse mappings on GFX7-8 due to GPU hangs in the VK CTS,
+    * except Polaris where it happens to work "well enough".
+    * Enable them when these are investigated and fixed in the driver.
     */
-   info->has_sparse_vm_mappings = info->gfx_level >= GFX7;
+   info->has_sparse_vm_mappings = info->family >= CHIP_POLARIS10;
    info->has_gang_submit = info->drm_minor >= 49;
    info->has_gpuvm_fault_query = info->drm_minor >= 55;
    info->has_tmz_support = device_info.ids_flags & AMDGPU_IDS_FLAGS_TMZ;
-- 
GitLab

